tests/core/actions/utils.py

```diff
@@ -0,0 +1,16 @@
+import json
+
+
+def validate_action_str(action_str, action_parameters):
+    """
+    Parse the action string as JSON within an exec call and verify that it matches the original action parameters.
+    Note that we need the exec call to mimic the same behavior in the rules container.
+
+    If this triggers a syntax error, the exec call is set up incorrectly and needs to be fixed.
+    If this triggers a JSON decode error, the action string is badly formatted.
+    If this triggers an assertion error, the deserialized action JSON does not match the original action parameters.
+    """
+    try:
+        exec(f'import json; assert json.loads("{action_str}") == {action_parameters}')
+    except (SyntaxError, json.JSONDecodeError, AssertionError) as e:
+        raise Exception(f"Error occurred during action string validation.")
```

Comment:
Already discussed here: https://github.com/awslabs/sagemaker-debugger-rulesconfig/pull/32#discussion_r545401362

TL;DR can't get the traceback since the call was made in exec. All we can get is the type of exception (unless we put in more work to capture output). Docstring has complete information about why validation could fail.

Reply:
any reason you wouldn't `raise..from`?

or at the very least include the exception `e` in the newly raised exception? as it stands, the `f-string` isn't used, and `e` isn't used.

Reply:
We are using the `e`, see [here](https://github.com/awslabs/sagemaker-debugger-rulesconfig/pull/32/files#diff-056dd7e56fcbd0d7105a6b9c7c2b8b321e8f191254456c20168b34d0cf1783e3R51).

I think you might've reviewed before I pushed this change.